### PR TITLE
Multiple Bug Fixes.

### PR DESCRIPTION
**Pull request by TheDecepTicon30**
Multiple Bug Fixes.
Description:

Following Bugs were fixed.

BugID     Bug Description
2619445 - nsxt_transport_nodes modules needs updates based on new spec
2648943 - updation of only tags is not supported by 'nsxt_transport_nodes' module
2825054 - Create Edge transport node using ansible module fails
2731783 - Getting "info from vCenter: [SSL: CERTIFICATE_VERIFY_FAILED]" when deploying 2nd and 3rd NSX-T managers
2643929 - "tags" field is not support by "nsxt_transport_node_profiles" module
2645685 - "tags" field is not supported by "nsxt_transport_node_collections" module
2645684 - Updating 'Description' and 'Transport Node Profile' fields in "nsxt_transport_node_collections" module fails
2619365 - nsxt_transport_node_profiles.py fails if transport_zone_endpiont is not provided outside of host_switches
2632169 - [Ansible]deletion of all 'lags' and creation of  'lags' is not supported by 'nsxt_uplink_profiles' ansible module

### CODE REVIEW

File: plugins/module_utils/vcenter_utils.py [hunk 1]
```diff
@@ -15,96 +15,116 @@
...
-                                                user=username,
-                                                pwd=password,
-                                                port=443)
-        if not service_instance:
-            module.fail_json(msg="Could not connect to the specified vCenter "
-                  "host using specified username and password")
-
-        atexit.register(connect.Disconnect, service_instance)
-    except (requests.ConnectionError, ssl.SSLError):
+    Retrieves vCenter information from service instance and returns as content object.
```
Comment (TheDecepTicon30) [inline plugins/module_utils/vcenter_utils.py:30]: Bug Fixed - 2731783 - Getting "info from vCenter: [SSL: CERTIFICATE_VERIFY_FAILED]" when deploying 2nd and 3rd NSX-T managers

Logic - 
Added a new flag ignore_ssl_connection to decide whether to use SSL while connecting to VC or not.

File: plugins/modules/nsxt_transport_node_collections.py [hunk 2]
```diff
@@ -192,7 +191,27 @@ def check_for_update(module, manager_url, mgr_username, mgr_password, validate_c
     if existing_tnc is None:
         return False
     if existing_tnc['compute_collection_id'] == transport_node_collection_with_ids['compute_collection_id'] and \
-        existing_tnc['transport_node_profile_id'] != transport_node_collection_with_ids['transport_node_profile_id']:
+            existing_tnc['transport_node_profile_id'] != \
```
Comment (TheDecepTicon30) [inline plugins/modules/nsxt_transport_node_collections.py:194]: Bugs Fixed - 
2645685 - "tags" field is not supported by "nsxt_transport_node_collections" module
2645684 - Updating 'Description' and 'Transport Node Profile' fields in "nsxt_transport_node_collections" module fails

File: plugins/modules/nsxt_transport_node_profiles.py [hunk 3]
```diff
@@ -254,6 +249,12 @@ def check_for_update(module, manager_url, mgr_username, mgr_password, validate_c
         transport_node_profile_with_ids.__contains__('host_switch_spec') and transport_node_profile_with_ids['host_switch_spec'].__contains__('host_switches') and \
         existing_transport_node_profile['host_switch_spec']['host_switches'] != transport_node_profile_with_ids['host_switch_spec']['host_switches']:
         return True
```
Comment (TheDecepTicon30) [inline plugins/modules/nsxt_transport_node_profiles.py:251]: Bugs Fixed - 2643929 - "tags" field is not support by "nsxt_transport_node_profiles" module

File: plugins/modules/nsxt_transport_nodes.py [hunk 4]
```diff
@@ -514,10 +522,6 @@
                       'present' is used to create or update resource. 
                       'absent' is used to delete resource."
         required: true
-    transport_zone_endpoints:
```
Comment (TheDecepTicon30) [inline plugins/modules/nsxt_transport_nodes.py:517]: Bugs Fixed - 
2619445 - nsxt_transport_nodes modules needs updates based on new spec
2648943 - updation of only tags is not supported by 'nsxt_transport_nodes' module

File: plugins/modules/nsxt_uplink_profiles.py [hunk 5]
```diff
@@ -226,6 +226,10 @@ def check_for_update(module, manager_url, mgr_username, mgr_password, validate_c
     if existing_profile.__contains__('transport_vlan') and profile_params.__contains__('transport_vlan') and \
         existing_profile['transport_vlan'] != profile_params['transport_vlan']:
         return True
+    if existing_profile.__contains__('lags') and not profile_params.__contains__('lags'):
```
Comment (TheDecepTicon30) [inline plugins/modules/nsxt_uplink_profiles.py:229]: Bugs Fixed - 2632169 - [Ansible]deletion of all 'lags' and creation of  'lags' is not supported by 'nsxt_uplink_profiles' ansible module

File: tests/playbooks/mp/test_transport_node_profiles.yml [hunk 6]
```diff
@@ -19,7 +19,6 @@
         host_switch_spec:
           resource_type: StandardHostSwitchSpec
           host_switches: "{{item.host_switches}}"
-          transport_zone_endpoints: "{{item.transport_zone_endpoints}}"
```
Comment (TheDecepTicon30) [inline tests/playbooks/mp/test_transport_node_profiles.yml:22]: Bug Fixed - 2619365 - nsxt_transport_node_profiles.py fails if transport_zone_endpiont is not provided outside of host_switches

File: tests/playbooks/mp/test_transport_nodes.yml [hunk 7]
```diff
@@ -17,7 +17,6 @@
         host_switch_spec:
           resource_type: StandardHostSwitchSpec
           host_switches: "{{item.host_switches}}"
-        transport_zone_endpoints: "{{item.transport_zone_endpoints}}"
```
Comment (TheDecepTicon30) [inline tests/playbooks/mp/test_transport_nodes.yml:20]: Bug Fixed - 2825054 - Create Edge transport node using ansible module fails - transport_zone_endpoints removed

**Comment by kaushiklele** [comment]
Pull request looks good to me. It can be approved.

**Pull request merged**